src/compiler/checker.ts

```diff
@@ -4979,16 +4983,26 @@ namespace ts {
             return checkTypeRelatedTo(source, target, assignableRelation, errorNode, headMessage, containingMessageChain);
         }
 
+        function isSignatureAssignableTo(source: Signature,
+                                         target: Signature,
+                                         ignoreReturnTypes: boolean): boolean {
+            return compareSignaturesRelated(source, target, ignoreReturnTypes, /*errorReporter*/ undefined, compareTypesAssignable) !== Ternary.False;
+        }
+
         /**
          * See signatureRelatedTo, compareSignaturesIdentical
          */
-        function isSignatureAssignableTo(source: Signature, target: Signature, ignoreReturnTypes: boolean): boolean {
+        function compareSignaturesRelated(source: Signature,

```

Comment:
But all the `isX` functions nested in checkTypeRelatedTo return Ternary. I would just call them `isSignatureIdenticalTo`, `isSignatureRelatedTo`, and `isSignatureAssignableTo`. I don't think the prefix `compare` is illuminating in any way. Then if there are any further distinctions to be made, they can be encoded more explicitly into the name, and consistently across all of these functions.